### ISSUE TITLE
Add auto-instrumentation for asyncpg for configure_azure_monitor

### ISSUE THREAD

**Issue by pamelafox** [Azure/azure-sdk-for-python#36583]
Since the auto-instrumentation already works for FastAPI and psycopg2, developers might be surprised to see that asyncpg is not supported. It'd be great if it was, since FastAPI + asyncpg will likely be an increasingly common combo, especially once SQLModel supports async. I am currently using it via SQLAlchemy's async support.

**Comment by swathipil** [comment]
Hi @pamelafox - Is there a specific package you're referring to?

**Comment by pamelafox** [comment]
This request is for azure-monitor-opentelemetry

**Comment by swathipil** [comment]
Hi @lzchen - would you be able to take a look?

**Comment by lzchen** [comment]
@swathipil 

Yes @pamelafox  and I have discussed offline. This is simply a feature request.
